pkg/cloud/amazon/aws_kms.go

```diff
@@ -128,38 +129,36 @@ func MakeAWSKMS(ctx context.Context, uri string, env cloud.KMSEnv) (cloud.KMS, e
 	}
 
 	region := kmsURIParams.region
-	awsConfig := &aws.Config{
-		Credentials: credentials.NewStaticCredentials(kmsURIParams.accessKey,
-			kmsURIParams.secret, kmsURIParams.tempToken),
+	var loadOptions []func(options *config.LoadOptions) error
+	addLoadOption := func(option config.LoadOptionsFunc) {
+		loadOptions = append(loadOptions, option)
 	}
-	awsConfig.Logger = newLogAdapter(ctx)
+	addLoadOption(config.WithLogger(newLogAdapter(ctx)))
 	if kmsURIParams.verbose {
-		awsConfig.LogLevel = awsVerboseLogging
+		addLoadOption(config.WithClientLogMode(awsVerboseLogging))
 	}
 
+	var endpointURI string
 	if kmsURIParams.endpoint != "" {
 		if env.KMSConfig().DisableHTTP {
 			return nil, errors.New(
 				"custom endpoints disallowed for aws kms due to --aws-kms-disable-http flag")
 		}
-		awsConfig.Endpoint = &kmsURIParams.endpoint
-		if region == "" {
```

Comment:
FWIW, we document that you need to set region: https://www.cockroachlabs.com/docs/stable/take-and-restore-encrypted-backups.

But I wonder if we should (as a follow up), (1) make sure we actually respect the AWS_REGION environment variable and (2) understand be the behaviour of `default-region` in the past to see if there is a cheap way to avoid breaking KMS URLS that are currently working.